clustermesh/clustermesh.go

```diff
@@ -1810,3 +1810,131 @@ func DisableWithHelm(ctx context.Context, k8sClient *k8s.Client, params Paramete
 	_, err = helm.Upgrade(ctx, k8sClient.RESTClientGetter, upgradeParams)
 	return err
 }
+
+// ConnectWithHelm enables clustermesh using a Helm Upgrade action Certificates are generated via
+// the Helm chart's cronJob (certgen) mode As with classic mode, only autodetected IP-based
+// clustermesh-apiserver Service endpoints are currently supported.
+func (k *K8sClusterMesh) ConnectWithHelm(ctx context.Context) error {
+	v, err := k.client.GetRunningCiliumVersion(ctx, k.params.Namespace)
```

Comment:
let's use version from helm chart instead of calling this function. it is a very shady function: https://github.com/cilium/cilium-cli/blob/main/k8s/client.go#L928-L944